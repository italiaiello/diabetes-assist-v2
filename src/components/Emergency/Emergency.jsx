import React from 'react'
import Phone from '../../images/ringing_phone.svg'

const Emergency = props => {
    return (
        <article className="pageDisplay">
            <div className="backAndHeading">
                    <div className="sectionHeading">
                        <div className="back">
                            <div className="leftArrow" onClick={() => props.onRouteChange('home')}></div>
                            <p>Back</p>
                        </div>
                        <h1>Emergency</h1>
                    </div>
                </div>
            <figure id="phone">
                <img src={Phone} alt="Phone ringing" />
            </figure>
            <p id="calling">Calling "<span>000</span>" For You...</p>
            <button id="cancelCall" onClick={() => props.onRouteChange('home')}>Cancel</button>
        </article>
    )
}

export default Emergency