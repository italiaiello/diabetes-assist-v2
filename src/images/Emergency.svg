<svg width="55" height="53" viewBox="0 0 55 53" fill="none" xmlns="http://www.w3.org/2000/svg">
<g filter="url(#filter0_d)">
<path d="M45.4812 0H16.7574C13.7144 0 11.2388 2.19427 11.2388 4.89137V23.8979C11.2388 26.595 13.7144 28.7892 16.7574 28.7892H35.351L45.2555 34.2819C45.3765 34.3489 45.5133 34.3822 45.65 34.3822C45.8053 34.3822 45.9601 34.3392 46.0908 34.2539C46.3365 34.0938 46.4468 33.8168 46.3693 33.5554L44.9552 28.7892H45.4813C48.5243 28.7892 50.9999 26.595 50.9999 23.8979V4.89137C50.9998 2.19427 48.5242 0 45.4812 0Z" fill="#E7E7F2"/>
<path d="M45.4814 0H42.2422C45.2852 0 47.7608 2.19427 47.7608 4.89137V23.8979C47.7608 26.595 45.2852 28.7892 42.2422 28.7892H41.7161L42.9696 33.0141L45.2557 34.2819C45.3767 34.349 45.5136 34.3823 45.6502 34.3823C45.8055 34.3823 45.9604 34.3393 46.0911 34.254C46.3367 34.0939 46.447 33.8169 46.3695 33.5555L44.9554 28.7893H45.4815C48.5245 28.7893 51.0001 26.595 51.0001 23.8979V4.89137C51 2.19427 48.5244 0 45.4814 0Z" fill="#BCBCCE"/>
<path d="M38.3935 11.2918H34.6201V7.94726C34.6201 7.58313 34.2871 7.28809 33.8764 7.28809H28.3623C27.9516 7.28809 27.6186 7.58313 27.6186 7.94726V11.2918H23.8453C23.4345 11.2918 23.1016 11.5869 23.1016 11.951V16.8384C23.1016 17.2026 23.4345 17.4976 23.8453 17.4976H27.6186V20.8422C27.6186 21.2063 27.9516 21.5014 28.3623 21.5014H33.8764C34.2871 21.5014 34.6201 21.2063 34.6201 20.8422V17.4976H38.3935C38.8042 17.4976 39.1371 17.2026 39.1371 16.8384V11.951C39.1371 11.5869 38.8042 11.2918 38.3935 11.2918Z" fill="#C9345F"/>
<path d="M35.7058 39.2102L27.1527 30.027C27.0233 29.8881 26.8354 29.8022 26.6326 29.7893C26.4302 29.7764 26.23 29.8375 26.0795 29.9586L22.2935 33.0024C20.5156 31.5188 18.8551 29.6251 17.4652 27.4913C16.0752 25.3575 15.0552 23.1361 14.4947 21.0295L19.3618 19.6453C19.5553 19.5903 19.715 19.4674 19.8036 19.3053C19.8924 19.1433 19.9023 18.956 19.8313 18.7872L15.1354 7.63006C14.9965 7.30021 14.5894 7.12531 14.2115 7.2328L8.8067 8.77009C6.56194 9.40862 4.8807 11.0655 4.41912 13.0945C3.74286 16.0666 3.88713 19.5026 4.83628 23.0311C5.64671 26.0503 7.00231 29.0317 8.8655 31.8921C10.7287 34.7524 12.9638 37.2838 15.5085 39.4155C18.4812 41.9081 21.7663 43.7369 25.0086 44.7039C25.6757 44.9029 26.3652 44.9999 27.0507 44.9999C28.6396 44.9999 30.2077 44.4788 31.4275 43.4982L35.6319 40.1182C35.9262 39.8816 35.9587 39.4817 35.7058 39.2102Z" fill="#50D399"/>
<path d="M27.1524 30.0266C27.023 29.8877 26.8351 29.8017 26.6323 29.7888C26.43 29.7759 26.2298 29.8371 26.0792 29.9581L23.9475 31.6719L33.0875 42.163L35.6317 40.1176C35.926 39.8811 35.9585 39.4813 35.7057 39.2097L27.1524 30.0266Z" fill="#32A873"/>
<path d="M16.6407 20.4189L19.3618 19.645C19.5553 19.59 19.715 19.4672 19.8036 19.3051C19.8924 19.143 19.9023 18.9557 19.8313 18.787L15.1354 7.63006C14.9965 7.30021 14.5895 7.12531 14.2115 7.2328L11.019 8.1408L16.6407 20.4189Z" fill="#32A873"/>
</g>
<defs>
<filter id="filter0_d" x="0" y="0" width="55.0001" height="52.9999" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB">
<feFlood flood-opacity="0" result="BackgroundImageFix"/>
<feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"/>
<feOffset dy="4"/>
<feGaussianBlur stdDeviation="2"/>
<feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"/>
<feBlend mode="normal" in2="BackgroundImageFix" result="effect1_dropShadow"/>
<feBlend mode="normal" in="SourceGraphic" in2="effect1_dropShadow" result="shape"/>
</filter>
</defs>
</svg>
