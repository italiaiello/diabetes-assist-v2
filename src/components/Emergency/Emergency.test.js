import { shallow } from 'enzyme';
import React from 'react';
import Emergency from './Emergency';

let route = ""

const mockOnRouteChange = (newRoute) => {
  route = newRoute;
}

describe('Emergency.js', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Emergency onRouteChange={mockOnRouteChange} />)
  })

  it('expects to render Emergency component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('checks that onRouteChange works onClick', () => {
    wrapper.find('[id="cancelCall"]').simulate('click')
    expect(route).toEqual('home')

    route = "chicken"

    wrapper.find('[className="leftArrow"]').simulate('click')
    expect(route).toEqual('home')
  })
}) 