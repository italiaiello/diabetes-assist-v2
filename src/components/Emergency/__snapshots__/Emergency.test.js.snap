// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emergency.js expects to render Emergency component 1`] = `ShallowWrapper {}`;
